frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx

```diff
@@ -30,32 +33,71 @@ const TopConsumerResourceValueMapping = {
   Pods: 'pod',
 };
 
-const metricTypes = _.keys(TopConsumerResourceValue);
-const sortByTypes = _.keys(TopConsumerSortByValue);
+const getTopConsumersQueriesMap = (
+  pluginItem: DashboardStorageExtension,
+  topConsumers: TopConsumersQueries,
+  category: string,
+): TopConsumersQueries => {
+  const { name } = pluginItem.properties;
+  const queryName = `${name}_${category}`;
+  if (!topConsumers[queryName]) {
+    TopConsumerResourceValue[name] = _.replace(queryName, category, '');
+    TopConsumerResourceValueMapping[name] = pluginItem.properties.metricType;
+    topConsumers[queryName] = pluginItem.properties.query;
+  }
+  return topConsumers;
+};
 
-const metricTypesOptions = _.zipObject(metricTypes, metricTypes);
-const sortByOptions = _.zipObject(sortByTypes, sortByTypes);
+const getItems = (plugin: DashboardStorageExtension[], flags: FlagsObject) =>
```

Comment:
Nit: the `plugin` parameter should be better named, it's actually an array of specific extension objects.

I will post a follow-up PR to address this.